### PR TITLE
Modifies testHook to return unmount function

### PR DESCRIPTION
**Pull request by Andrewmat**
This PR solves #289, in which I asked to have a way to test the `useEffect` callback.

**What**:
Adds a way to test hooks behavior when component unmounts.

**Why**:
To test `useEffect` callbacks, which are called when component unmounts.

**How**:
Modifies `testHook` to return an object with an exposed `unmount` function in it.
As said in the issue #289, I didn't expose anything else to avoid encouraging unfocused tests. In the future, if someone asks for them, they could be included.

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged
- [x] Added myself to contributors table

I also modified the file `examples/react-hooks` so `useCounter` is not a default export, and added `useDocumentTitle` as example.
I am not confident about the `useDocumentTitle` though, as its implementation is quirky and not a good example (*ironically*). But I couldn't think of another simple use of `useEffect` and its callback.

### CODE REVIEW

File: examples/__tests__/react-hooks.js [hunk 1]
```diff
@@ -6,53 +6,76 @@
  */
```
Comment (kentcdodds) [inline examples/__tests__/react-hooks.js]: I'm guessing you had to skip the pre-commit hook or something. This should have auto-formatted when you commit it. Could you run `npx kcd-scripts format` please?

File: src/index.js [hunk 2]
```diff
@@ -67,7 +67,8 @@ function TestHook({callback}) {
 }
 
 function testHook(callback) {
-  render(<TestHook callback={callback} />)
+  const { unmount } = render(<TestHook callback={callback} />)
```
Comment (kentcdodds) [inline src/index.js]: Let's also return `rerender` as well I think.

File: examples/__tests__/react-hooks.js (hunk 1 above)
Comment (Andrewmat) [reply examples/__tests__/react-hooks.js]: I ran it and it added a linebreak in `ISSUE_TEMPLATE.md` and `mock.react-transition-group.js`. Should I also add it to the commit?
captura de tela de 2019-02-10 12-48-41
captura de tela de 2019-02-10 12-48-23

**Comment by FredyC** [comment]
Would be nice to update typescript definition when changing API. 🙊

**Comment by Andrewmat** [comment]
I haven't mess with TS typing for a long time now. Can you check if this is correct?
https://github.com/Andrewmat/react-testing-library/blob/master/typings/index.d.ts

**Comment by kentcdodds** [comment]
Looks correct to me.

**Comment by Andrewmat** [comment]
Should I also add some guidance inside CONTRIBUTING.md?
```
If your PR introduced some changes in the API, you are more than welcome to modify the Typescript type definition to reflect those changes. Just modify the `/typings/index.d.ts` file accordingly. If you have never seen Typescript definitions before, you can read more about it in their [documentation pages](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html)
```
I thought about adding a new section in "Commiting and Pushing changes"

**Comment by kentcdodds** [comment]
Sure :+1:

**Comment by thealjey** [comment]
wow, mind absolutely blown

I love this library

**Pull request merged**